nautobot/extras/tests/test_models.py

```diff
@@ -1324,6 +1327,252 @@ def get_absolute_url(self):
         self.assertEqual("", log.absolute_url)
 
 
+class ObjectMetadataTest(ModelTestCases.BaseModelTestCase):
+    model = ObjectMetadata
+
+    def test_immutable_metadata_type(self):
+        instance1 = ObjectMetadata.objects.first()
+        instance2 = ObjectMetadata.objects.exclude(metadata_type=instance1.metadata_type).first()
+        self.assertIsNotNone(instance2)
+        with self.assertRaises(ValidationError):
+            instance1.metadata_type = instance2.metadata_type
+            instance1.validated_save()
+
+    def test_contact_team_mutual_exclusive(self):
```

Comment:
In general I think we need positive tests (valid values) for each datatype as well?

Reply:
Included type conversion behaviors
1. from int to float
2. from float to int
3. from String to int
4. from String to float